server/src/main/scala/slickdb/UserTable.scala

```diff
@@ -16,19 +16,47 @@ case class UserRow(
                     inactive_email_sent: Option[java.sql.Timestamp],
                     revoked_access: Option[java.sql.Timestamp])
 
+trait RateLimit {
+  private val rateLimitDuration: FiniteDuration = 1.minute
+  private val maxCallsPerUser: Int = 10
+
+  private val userCalls = new scala.collection.mutable.HashMap[String, (Int, Long)]()
+
+  protected def limitRate[A](userId: String)(action: => Future[A])(implicit ec: ExecutionContext): Future[A] = {
+    synchronized {
+      val (userCallCount, lastCallTime): (Int, Long) = userCalls.getOrElseUpdate(userId, (0, System.currentTimeMillis()))
+      val currentTime: Long = System.currentTimeMillis()
+      val elapsedTime = currentTime - lastCallTime
+
+      if (elapsedTime >= rateLimitDuration.toMillis) {
+        // Reset the call count if the rate limit duration has passed since the last call
+        userCalls.update(userId, (0, currentTime))
+      }
+      val updatedCallCount = userCalls.getOrElseUpdate(userId, (0, currentTime))._1
+      if (updatedCallCount < maxCallsPerUser) {
+        userCalls.update(userId, (updatedCallCount + 1, currentTime))
+        action.andThen { case _ => userCalls.update(userId, (updatedCallCount, currentTime)) }
+      } else {
+        Future.failed(new RuntimeException("Rate limit exceeded"))
+      }
+    }
+  }
+}
+
 trait UserTableLike {
 
   def selectAll: Future[Seq[UserRow]]
 
+  def selectUser(email: String)(implicit ec: ExecutionContext): Future[Option[UserRow]]
```

Comment:
I don't think this function is used. Is it from when you tried storing the views in the user table?